File: php/examples/geometric_mean.php
<?php

require_once('../lib/mean_averages.php');

$sampleData = fopen('../../sample_data', 'r');
$sampleIndex = 1;
while (!feof($sampleData)) {
    $line = fgets($sampleData);
    if (empty(trim($line))) {
        continue;
    }

    $sample = explode(',', $line);
    array_walk($sample, function(&$point) {
        $point = floatval($point);
    });

    $average = ClounceMath\MeanAverages::geometricMean($sample);
    
    print("Sample $sampleIndex" . PHP_EOL);
    print('Sample Data: ' . implode(', ', $sample) . PHP_EOL);
    print("Geometric Mean: $average" . PHP_EOL);
    print(PHP_EOL . PHP_EOL);
    $sampleIndex++;
}

File: python/examples/geometric_mean.py
import sys

sys.path.insert(0, '../lib/')
from mean_averages import MeanAverages

if __name__ == "__main__":
    sample_data = open('../../sample_data', 'r')

    sample_index = 1
    for line in sample_data:
        if not line or line.isspace():
            continue

        print('Sample ', sample_index)
        
        sample = [ float(data) for data in line.split(',') ]
        print('Sample Data: ', sample)
        print('Geometric Mean: ', MeanAverages.geometric_mean(sample))

        sample_index += 1
        print()
        print()
    
    sample_data.close()
